client/src/project/components/GitLabConnect.tsx

```diff
@@ -0,0 +1,64 @@
+import React from "react";
+import { DropdownItem } from "reactstrap";
+
+import { ExternalLink } from "../../utils/components/ExternalLinks";
+import { ButtonWithMenu } from "../../utils/components/buttons/Button";
+
+function externalUrlToGitLabIdeUrl(externalUrl: string) {
+  if (externalUrl.includes("/gitlab/")) return externalUrl.replace("/gitlab/", "/gitlab/-/ide/project/");
+  const url = new URL(externalUrl);
+  const pathname = url.pathname;
+  const newPathname = `/-/ide/project${pathname}`;
+  url.pathname = newPathname;
+  return url.toString();
+}
+
+type GitLabLinkItemProps = {
+  size: string;
+  text: string;
+  url: string;
+};
+
+function GitLabLinkItem({ size, text, url }: GitLabLinkItemProps) {
+  const onClick = () => window.open(url, "_blank");
+  return (
+    <DropdownItem onClick={onClick} size={size}>
+      {text}
+    </DropdownItem>
+  );
```

Comment:
`window.open` doesn't allow the typical interactions offered by an anchor element.
In other similar scenarios, we use the `ExternalLink` component with the `nav-link` class. I suggest adopting the same solution here

```suggestion
  return (
    <DropdownItem onClick={onClick} size={size}>
      <ExternalLink className="nav-link" role="text" title={text} url={url} />
    </DropdownItem>
  );
```

Reply:
Good suggestion, I have incorporated it.